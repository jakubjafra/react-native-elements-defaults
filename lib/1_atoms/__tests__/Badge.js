import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

describe('with no theme applied', () => {
    let OldBadge, Badge;

    beforeAll(() => {
        let defaultStyles = {};

        OldBadge = require('react-native-elements/src/badge/badge').default;
        Badge = require('../Badge').compose(defaultStyles)(OldBadge);
    });

    test('OldBadge and Badge should be different', () => {
        expect(OldBadge).not.toBe(Badge);
    });

    test('OldBadge renders correctly', () => {
        const tree = renderer.create(<OldBadge>Badge</OldBadge>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Badge renders correctly', () => {
        const tree = renderer.create(<Badge>Badge</Badge>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('OldBadge and Badge returns the same thing if no theme is applied', () => {
        const oldTree = renderer.create(<OldBadge>Badge</OldBadge>).toJSON();
        const tree = renderer.create(<Badge>Badge</Badge>).toJSON();

        expect(JSON.stringify(oldTree)).toEqual(JSON.stringify(tree));
    });
});

describe('with basic theme applied', () => {
    let OldBadge, Badge;

    beforeAll(() => {
        let defaultStyles = {
            containerStyle: {
                fontFamily: 'Arial',
            }
        };

        OldBadge = require('react-native-elements/src/badge/badge').default;
        Badge = require('../Badge').compose(defaultStyles)(OldBadge);
    });

    test('OldBadge and Badge should be different', () => {
        expect(OldBadge).not.toBe(Badge);
    });

    test('OldBadge renders correctly', () => {
        const tree = renderer.create(<OldBadge>Badge</OldBadge>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Badge renders correctly', () => {
        const tree = renderer.create(<Badge>Badge</Badge>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('OldBadge and Badge should be different caused by theme applied', () => {
        const oldTree = renderer.create(<OldBadge>Badge</OldBadge>).toJSON();
        const tree = renderer.create(<Badge>Badge</Badge>).toJSON();

        expect(oldTree).not.toEqual(tree);
    });
});
