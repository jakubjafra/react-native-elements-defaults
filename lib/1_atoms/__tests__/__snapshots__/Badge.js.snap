// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with basic theme applied Badge renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View
    onPress={undefined}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#43484d",
          "borderRadius": 20,
          "justifyContent": "center",
          "padding": 12,
          "paddingBottom": 3,
          "paddingTop": 3,
        },
        Object {
          "fontFamily": "Arial",
        },
      ]
    }
  >
    Badge
  </View>
</View>
`;

exports[`with basic theme applied OldBadge renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View
    onPress={undefined}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#43484d",
          "borderRadius": 20,
          "justifyContent": "center",
          "padding": 12,
          "paddingBottom": 3,
          "paddingTop": 3,
        },
        undefined,
      ]
    }
  >
    Badge
  </View>
</View>
`;

exports[`with no theme applied Badge renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View
    onPress={undefined}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#43484d",
          "borderRadius": 20,
          "justifyContent": "center",
          "padding": 12,
          "paddingBottom": 3,
          "paddingTop": 3,
        },
        undefined,
      ]
    }
  >
    Badge
  </View>
</View>
`;

exports[`with no theme applied OldBadge renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View
    onPress={undefined}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#43484d",
          "borderRadius": 20,
          "justifyContent": "center",
          "padding": 12,
          "paddingBottom": 3,
          "paddingTop": 3,
        },
        undefined,
      ]
    }
  >
    Badge
  </View>
</View>
`;
